src/ouroboros_gt/gt_descriptors.py

```diff
@@ -1,17 +1,21 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from ouroboros.config import Config, register_config
 import numpy as np
-
 import ouroboros as ob
 
 
-def get_gt_descriptor_model():
-    return GtDescriptorModel()
-
-
 class GtDescriptorModel:
-    def __init__(self):
+    def __init__(self, config: GtDescriptorModelConfig):
         pass
 
     def infer(self, image: ob.SparkImage, keypoints: np.ndarray, pose_hint: ob.VlcPose):
         if pose_hint is None:
             raise Exception("GtDescriptorModel requires setting pose_hint")
         return np.array([pose_hint.to_descriptor()])
+
+
+@register_config("descriptor_model", name="ground_truth", constructor=GtDescriptorModel)
+@dataclass
+class GtDescriptorModelConfig(Config):
```

Comment:
More a note for myself, but it's kinda annoying to have to specify empty configs for the factory functions (I think it's possible to manually register a constructor that doesn't bother instantiating the config)